src/components/vm/disks/diskAdd.jsx

```diff
@@ -513,11 +514,49 @@ export class AddDiskModalBody extends React.Component {
             break;
         }
         case 'file': {
+            this.setState({ file: value });
+
             if (value.endsWith(".iso")) {
                 // use onValueChange instead of setState in order to perform subsequent state change logic
                 this.onValueChanged("device", "cdrom");
             }
-            this.setState({ file: value });
+
+            if (value) {
+                cockpit.spawn(["/bin/bash", "-c", "xargs -0n1 head --bytes=16 "], { binary: true, err: "message", superuser: "try" })
+                        .input(value)
+                        .then(output => {
+                            console.log(output);
+                            let format = "";
+                            const file_identifier = Array.from(output).slice(0, 4)
```

Comment:
Please save Array.from(output) in a const, as you use it multiple times. But is this actually necessary? You already get an `Uint8Array` from spawn, which seems good enough?

Reply:
I find it kinda annoying to match against, but I guess I can change that.